dissect/target/plugins/os/unix/log/messages.py

```diff
@@ -96,18 +97,23 @@ def _parse_cloud_init_log(self, log_file: Path) -> Iterator[MessagesRecord]:
 
         Returns: ``MessagesRecord``
         """
-        for line in log_file.open("rt").readlines():
-            if line := line.strip():
-                if match := RE_CLOUD_INIT_LINE.match(line):
-                    match = match.groupdict()
-                    yield MessagesRecord(
-                        ts=match["ts"].split(",")[0],
-                        daemon=match["daemon"],
-                        pid=None,
-                        message=match["message"],
-                        source=log_file,
-                        _target=self.target,
-                    )
-                else:
-                    self.target.log.warning("Could not match cloud-init log line")
+
+        with open_decompress(log_file, "rt") as fh:
+            for line in fh.readlines():
```

Comment:
```suggestion
            for line in fh:
```
Makes it lazily read lines too.